yjit/src/backend/ir.rs

```diff
@@ -534,71 +534,6 @@ impl Assembler
         asm
     }
 
-    /// Transforms the instructions by splitting instructions that cannot be
-    /// represented in the final architecture into multiple instructions that
-    /// can.
-    pub(super) fn split_loads(self) -> Assembler
-    {
-        // Load operands that are GC values into a register
-        fn load_gc_opnds(op: Op, opnds: Vec<Opnd>, asm: &mut Assembler) -> Vec<Opnd>
-        {
-            if op == Op::Load || op == Op::Mov {
-                return opnds;
-            }
-
-            fn map_opnd(opnd: Opnd, asm: &mut Assembler) -> Opnd {
-                if let Opnd::Value(val) = opnd {
-                    // If this is a heap object, load it into a register
-                    if !val.special_const_p() {
-                        asm.load(opnd);
-                    }
-                }
-
-                opnd
-            }
-
-            opnds.into_iter().map(|opnd| map_opnd(opnd, asm)).collect()
-        }
```

Comment:
Did you test the results of the changes to the x86 codegen? I would have preferred that this be part of a separate PR (or that I make this change on my end).

Reply:
We can revert the last commit if you'd like and pull it in later. Yes, I tested it against the x86 codegen, from what I could tell it generated the same code. All of the tests continued to pass as well.

Reply:
Sorry I didn't realize you wanted to write this code, I didn't mean to overstep here.